posthog/temporal/batch_exports/http_batch_export.py

```diff
@@ -0,0 +1,251 @@
+
+        logger.info("BatchExporting %s rows", count)
+
+        if inputs.batch_export_schema is not None:
+            raise NotImplementedError("Batch export schema is not supported for HTTP export")
+
+        fields = http_default_fields()
+        columns = [field["alias"] for field in fields]
+
+        record_iterator = iter_records(
+            client=client,
+            team_id=inputs.team_id,
+            interval_start=inputs.data_interval_start,
+            interval_end=inputs.data_interval_end,
+            exclude_events=inputs.exclude_events,
+            include_events=inputs.include_events,
+            fields=fields,
+            extra_query_parameters=None,
+        )
+
+        rows_exported = get_rows_exported_metric()
+        bytes_exported = get_bytes_exported_metric()
+
+        # The HTTP destination currently only supports the PostHog batch capture endpoint. In the
+        # future we may support other endpoints, but we'll need a way to template the request body,
+        # headers, etc.
+        #
+        # For now, we write the batch out in PostHog capture format, which means each Batch Export
+        # temporary file starts with a header and ends with a footer.
+        #
+        # For example:
+        #
+        #   Header written when temp file is opened: {"api_key": "api-key-from-inputs","batch": [
+        #   Each record is written out as an object:   {"event": "foo", ...},
+        #   Finally, a footer is written out:        ]}
+        #
+        # Why write to a file at all? Because we need to serialize the data anyway, and it's the
+        # safest way to stay within batch endpoint payload limits and not waste process memory.
+        posthog_batch_header = """{"api_key": "%s","batch": [""" % inputs.token
+        posthog_batch_footer = "]}"
+
+        with BatchExportTemporaryFile() as batch_file:
+
+            def write_event_to_batch(event):
+                if batch_file.records_since_last_reset == 0:
+                    batch_file.write(posthog_batch_header)
+                else:
+                    batch_file.write(",")
+
+                batch_file.write_record_as_bytes(json_dumps_bytes(event))
+
+            async def flush_batch():
+                logger.debug(
+                    "Sending %s records of size %s bytes",
+                    batch_file.records_since_last_reset,
+                    batch_file.bytes_since_last_reset,
+                )
+
+                batch_file.write(posthog_batch_footer)
+                await post_json_file_to_url(inputs.url, batch_file)
+
+                rows_exported.add(batch_file.records_since_last_reset)
+                bytes_exported.add(batch_file.bytes_since_last_reset)
+
+            for record_batch in record_iterator:
+                for row in record_batch.select(columns).to_pylist():
+                    # Format result row as PostHog event, write JSON to the batch file.
+
+                    properties = row["properties"]
```

Comment:
Also add `properties.$geoip_disable = true` we know for sure we want this here & it should actually also be in migration tools, not sure why it isn't. E.g. replicator has it: https://github.com/PostHog/posthog-plugin-replicator/blob/main/index.ts#L63
I also realised, once we have this we can remove replicator app completely.